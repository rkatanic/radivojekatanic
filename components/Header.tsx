import NavLink from "./NavLink";
import ThemeSwitch from "./ThemeSwitch";
import { FiMenu } from "react-icons/fi";
import Menu from "./Menu";

const LINKS = [
  { link: "/", label: "About" },
  { link: "/labs", label: "Labs" },
  { link: "/contact", label: "Contact" },
];

const Header = (): JSX.Element => (
  <div className="m-auto w-full max-w-2xl flex items-center justify-between py-10 font-medium">
    <div className="dark:text-gray-200 text-gray-900 text-sm">RK</div>
    <Menu />
  </div>
);

export default Header;
