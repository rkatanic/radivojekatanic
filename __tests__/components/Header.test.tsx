import Header from "@/components/Header";
import { render } from "@testing-library/react";
import { useRouter } from "next/router";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("Header", (): void => {
  it.only("should render", (): void => {
    (useRouter as jest.Mock).mockReturnValue({
      asPath: "/",
    });

    const { baseElement } = render(<Header />);

    expect(baseElement).toMatchSnapshot();
  });
});
