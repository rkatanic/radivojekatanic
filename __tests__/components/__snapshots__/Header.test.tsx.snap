// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<body>
  <div>
    <div
      class="m-auto w-full max-w-2xl flex items-center justify-between py-10 font-medium"
    >
      <div
        class="dark:text-gray-200 text-gray-900 text-sm"
      >
        RK
      </div>
      <div
        class="flex gap-4 items-center"
      >
        <span
          class="cursor-pointer dark:hover:text-gray-200 hover:text-gray-900 dark:text-gray-200 text-gray-900"
        >
          About
        </span>
        <span
          class="cursor-pointer dark:hover:text-gray-200 hover:text-gray-900 dark:text-gray-500 text-gray-400"
        >
          Labs
        </span>
        <span
          class="cursor-pointer dark:hover:text-gray-200 hover:text-gray-900 dark:text-gray-500 text-gray-400"
        >
          Contact
        </span>
        <div
          class="lg:hidden"
        >
          <button
            class="border w-9 h-9 flex items-center justify-center rounded-full bg-white dark:bg-gray-800 dark:border-gray-700 cursor-pointer group dark:hover:bg-gray-100 hover:bg-gray-900 shadow-md dark:shadow-black"
            type="button"
          >
            <svg
              class="dark:fill-gray-100 dark:group-hover:fill-black group-hover:fill-white"
              fill="currentColor"
              height="1rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2V4a8 8 0 1 0 0 16z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
